jina/serve/networking.py

```diff
@@ -178,6 +183,10 @@ def add_connection(self, address: str, deployment_name: str):
             stubs, channel = self._create_connection(address, deployment_name)
             self._address_to_channel[address] = channel
             self._connections.append(stubs)
+            # create a new set of stubs and channels for warmup to avoid
+            # loosing channel during remove_connection or reset_connection
+            stubs, _ = self._create_connection(address, deployment_name)
+            self._warmup_stubs.add(stubs)
```

Comment:
I do not see where they are used

Reply:
Check the usage of the property method `self.warmup_stubs`. What's the preference when exposing `read only` properties?

Reply:
ah okey